### PR TITLE
optv - 0.2.3

### PR DESCRIPTION
**Pull request by alexlib**
@conda-forge/help-python

### CODE REVIEW

**Comment by alexlib** [comment]
> A major thing on versioning. Happy to look at the rest when this is resolved!

sorry - I do not understand the issue. what's wrong with versioning?

**Comment by beckermr** [comment]
You cannot build the package from master. It has to be a released tarball. This way we can rebuild the package and get the same thing.

**Comment by dbast** [comment]
`git_tag: master` is not an accepted version as this is a branch that may change, etc.

think about something like:
> source:
>   url: https://github.com/OpenPTV/openptv/archive/v{{ version }}.tar.gz
>   sha256: ea7656717d36e936c2c64810e13f1c59cfaef5ba5feb367b2dc2e42828541029

File: recipes/liboptv/meta.yaml [hunk 1]
```diff
@@ -14,12 +14,18 @@ requirements:
     - {{ compiler('cxx') }}
     - python
     - setuptools
-    - numpy=1.16
+    - numpy x.x
     - cython
     - pyyaml
```
Comment (beckermr) [inline recipes/liboptv/meta.yaml]: I'd check out the docs here for how to build with numpy. https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/liboptv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: optv
+  version: 0.2.3
+
+source:
+  git_url: https://github.com/OpenPTV/openptv.git
+
+build:
+  number: 1
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, reset the build number

File: recipes/liboptv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: optv
+  version: 0.2.3
+
+source:
+  git_url: https://github.com/OpenPTV/openptv.git
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, do not use git, use tarballs instead
Please see:
https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=tarball#build-from-tarballs-not-repos

File: recipes/liboptv/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  script: cd py_bind && python setup.py prepare && python setup.py install
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - python
+    - setuptools
+    - numpy x.x
+    - cython
+    - pyyaml
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, the libraries which you need to build this project should be in ``host``. Following your requirements, that should be something like:
```
requirements:
  build:
    - {{ compiler('cxx') }}
  host:
    - python
    - setuptools
    - numpy
    - cython
    - pyyaml
  run:
    - {{ pin_compatible('numpy') }}
```

File: recipes/liboptv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+test:
+  requires:
+    - python
+    - setuptools
+    - - {{ pin_compatible('numpy') }}
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: there is an error here, you wrote two ``-``

Comment (marcelotrevisani) [reply recipes/liboptv/meta.yaml]: and I think your requirements to run this library should be in run requirements

File: recipes/liboptv/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: optv
+  version: 0.2.3
+
+source:
+  url: https://github.com/OpenPTV/openptv/archive/0.2.3.tar.gz
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, also add ``sha256``
Please see:
https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=sha256#populating-the-hash-field

File: recipes/liboptv/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+about:
+  home: http://www.openptv.net
+  license:  LGPL v3.0
+  summary: "A library of core algorithms for OpenPTV"
+  
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, remove the unnecessary comments

File: recipes/liboptv/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    - scipy
+    - nose
+  commands:
+    - nosetests
+  imports:
+    - optv
+
+about:
+  home: http://www.openptv.net
+  license:  LGPL v3.0
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, add the ``license_file``
Please see:
https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=license_file#about

File: recipes/liboptv/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    - python
+    - {{ pin_compatible('numpy') }}
+    - pyyaml
+
+test:
+  requires:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - pyyaml
+    - nose
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please remove these requirements from test
```
    - python
    - {{ pin_compatible('numpy') }}
    - pyyaml
```

File: recipes/liboptv/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  script: cd py_bind && python setup.py prepare && python setup.py install
+  
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - setuptools
+    - numpy
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: You need to also specify the numpy version which you need
In you project is hardcoded the version ``1.16.1``. So you should replace this line by
```
- numpy 1.16.1
```

File: recipes/liboptv/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  name: optv
+  version: 0.2.3
+
+source:
+  url: https://github.com/OpenPTV/openptv/archive/0.2.3.tar.gz
+  sha256: 21b8709dedc8696ff454d83d2d6af582252dbd9cf621f5d78ecdd64ce0af48cb
+
+build:
+  number: 0
+  script: cd py_bind && python setup.py prepare && python setup.py install
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please, change this line to:
```
script:
  - cd py_bind
  - {{ PYTHON }} setup.py prepare
  - {{ PYTHON }} -m pip install . --no-deps -vv
```

File: recipes/liboptv/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    - pip
+    - numpy 1.16.1
+    - cython
+    - pyyaml
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
+    - pyyaml
+
+test:
```
Comment (marcelotrevisani) [inline recipes/liboptv/meta.yaml]: Please add
```
files:
  - py_bind/test
```

**Comment by marcelotrevisani** [comment]
> thanks a lot @marcelotrevisani , but the nosetests has to run inside this directory, it reads plenty of hard-coded paths to the test files. shall I change it according to your structure?

That is one of the reasons which I prefer ``pytest`` 😆 

I already change it, let's see

File: recipes/liboptv/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,50 @@
+package:
```
Comment (chrisburr) [inline recipes/liboptv/meta.yaml]: Using a variable will allow the bot to automatically make update pull requests and ensure things don't get out of sync:
```suggestion
{% set version = "0.2.3" %}

package:
```

File: recipes/liboptv/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  name: optv
+  version: 0.2.3
```
Comment (chrisburr) [inline recipes/liboptv/meta.yaml]:
```suggestion
  version: {{ version }}
```

File: recipes/liboptv/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  url: https://github.com/OpenPTV/openptv/archive/0.2.3.tar.gz
```
Comment (chrisburr) [inline recipes/liboptv/meta.yaml]:
```suggestion
  url: https://github.com/OpenPTV/openptv/archive/{{ version }}.tar.gz
```

**Comment by marcelotrevisani** [comment]
Your tests are failing, you may have to fix them

**Comment by marcelotrevisani** [comment]
You8r code is not compatible with python 3, at least not your tests

One of the problems is that the tests are using the library ``mhlib`` which was removed. Consider to use ``mailbox``, please see

https://docs.python.org/2/library/mhlib.html

**Comment by marcelotrevisani** [comment]
If you still want to add this to ``conda-forge`` you have to skip Python 3

**Comment by alexlib** [comment]
going to fix tests and resubmit as a new version

**Pull request closed without merging**